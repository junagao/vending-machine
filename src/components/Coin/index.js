export { default as CoinItem } from './CoinItem';
export { default as CoinList } from './CoinList';
export { default as CoinSlot } from './CoinSlot';
export { default as CollectRefund } from './CollectRefund';
export { default as DisplayDepositedAmount } from './DisplayDepositedAmount';
export { default as RefillCoins } from './RefillCoins';
export { default as Wallet } from './Wallet';
