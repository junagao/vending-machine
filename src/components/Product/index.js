export { default as CollectProduct } from './CollectProduct';
export { default as ProductList } from './ProductList';
export { default as ProductItem } from './ProductItem';
export { default as RefillProductStock } from './RefillProductStock';
