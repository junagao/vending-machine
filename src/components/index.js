export { default as CoinSlot } from 'components/CoinSlot';
export { default as CollectProduct } from 'components/CollectProduct';
export { default as VendingMachine } from 'components/VendingMachine';
export { default as VendingMachineScreen } from 'components/VendingMachineScreen';
export { default as Wallet } from 'components/Wallet';
export { default as ProductList } from 'components/ProductList';
export { default as Header } from 'components/Header';
export { default as CollectRefund } from 'components/CollectRefund';
