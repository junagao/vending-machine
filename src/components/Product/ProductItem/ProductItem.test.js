import React from 'react';
import { shallow } from 'enzyme';
import { ProductItem } from 'components';

describe('ProductItem', () => {
  let wrapper;
  const idMock = 'product-a23d54';
  const nameMock = 'coca cola';
  const imgMock = 'http://vendingmachine.com/coke.png';
  const priceMock = 20;
  const currentStockMock = 7;
  const onSelectProductMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <ProductItem
        id={idMock}
        name={nameMock}
        img={imgMock}
        price={priceMock}
        currentStock={currentStockMock}
        onSelectProduct={onSelectProductMock}
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one li ProductItemLi element', () => {
    expect(wrapper.find('ProductItemLi').length).toEqual(1);
  });

  it('should render one ProductItemButton element of type `button`', () => {
    expect(wrapper.find('ProductItemButton').length).toEqual(1);
    expect(wrapper.find('ProductItemButton').prop('type')).toEqual('button');
  });

  it('should test click event on ProductItemButton button and pass `name, img, id` to onSelectProduct()', () => {
    wrapper.find('ProductItemButton').simulate('click');
    expect(onSelectProductMock.mock.calls[0][0]).toEqual('coca cola');
    expect(onSelectProductMock.mock.calls[0][1]).toEqual(
      'http://vendingmachine.com/coke.png',
    );
    expect(onSelectProductMock.mock.calls[0][2]).toEqual('product-a23d54');
    expect(onSelectProductMock.mock.calls.length).toEqual(1);
  });

  it('should render one img element with the correct `src, width, alt` props', () => {
    expect(wrapper.find('img').length).toEqual(1);
    expect(wrapper.find('img').prop('src')).toEqual(
      `http://vendingmachine.com/coke.png`,
    );
    expect(wrapper.find('img').prop('width')).toEqual(`46`);
    expect(wrapper.find('img').prop('alt')).toEqual(`coca cola`);
  });

  it('should render one ProductPrice p element with the price`', () => {
    expect(wrapper.find('ProductPrice').length).toEqual(1);
    expect(
      wrapper
        .find('ProductPrice')
        .children()
        .at(0)
        .text(),
    ).toEqual('20');
    expect(
      wrapper
        .find('ProductPrice')
        .children()
        .at(1)
        .text(),
    ).toEqual(' â‚¬');
  });

  it('should render one ProductQuantity element with the correct quantity amount. display `unit` if one or `units` is more than one', () => {
    expect(wrapper.find('ProductQuantity').length).toEqual(1);
    expect(
      wrapper
        .find('ProductQuantity')
        .children()
        .text(),
    ).toEqual(`${currentStockMock} units`);
    wrapper.setProps({ currentStock: 1 });
    expect(
      wrapper
        .find('ProductQuantity')
        .children()
        .text(),
    ).toEqual('1 unit');
  });
});
