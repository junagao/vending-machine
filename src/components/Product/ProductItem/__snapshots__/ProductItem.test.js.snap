// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem should render correctly given the required props 1`] = `
<li>
  <ProductItemButton
    onClick={[Function]}
    type="button"
  >
    <ProductImg
      alt="coca cola"
      src="http://vendingmachine.com/coke.png"
      width="46"
    />
    <ProductDetails>
      <ProductPrice>
        20
         â‚¬
      </ProductPrice>
      <ProductQuantity>
        7 units
      </ProductQuantity>
    </ProductDetails>
  </ProductItemButton>
</li>
`;
